src/backend/distributed/commands/statistics.c

```diff
@@ -230,7 +230,17 @@ DropStatisticsObjectAddress(Node *node, bool missing_ok, bool isPostprocess)
 												 dropStatisticsStmt->missing_ok);
 
 		ObjectAddress *objectAddress = palloc0(sizeof(ObjectAddress));
-		ObjectAddressSet(*objectAddress, StatisticExtRelationId, statsOid);
+
+		/*
+		 * we consider stat object as invalid if it is not owned by current user.
+		 * user should own the stat object to not diverge from pg output.
+		 */
+		if (OidIsValid(statsOid) && pg_statistics_object_ownercheck(statsOid,
```

Comment:
This is a little unusual to skip at this function. Two questions I have: 

- Can we skip non-owned statistics in another place? Like pre-process? Or is that too late to print the error?
- Why would we only hit this problem on `statistics` objects, not anything other?

Reply:
* We can move that check to PreprocessDropStatisticsStmt before qualifying the object there,
* There is a test case for statistics like below and it logs a qualified name error.
```sql
CREATE TABLE ab1 (a INTEGER, b INTEGER, c INTEGER);
CREATE STATISTICS IF NOT EXISTS ab1_a_b_stats ON a, b FROM ab1;
COMMENT ON STATISTICS ab1_a_b_stats IS 'new comment';
CREATE ROLE regress_stats_ext;
SET SESSION AUTHORIZATION regress_stats_ext;
DROP STATISTICS ab1_a_b_stats;```
ERROR:  must be owner of statistics object ab1_a_b_stats

Reply:
Maybe we can move it to preprocess if we set qualify method to NULL for the drop stats distops. We are already calling QualifyTreeNodein its preprocess. Then, we can explicitly call the qualify method for it instead of QualifyTreeNode.